source/fab/language/fortran.py

```diff
@@ -310,11 +310,10 @@ def analyse(self, source: TextReader) -> None:
                     unit_type: str = unit_match.group(1).lower()
                     unit_name: str = unit_match.group(2).lower()
                     logger.debug('Found %s called "%s"', unit_type, unit_name)
-                    self._state.add_fortran_program_unit(unit_name,
-                                                         source.filename)
+                    self._state.add_fortran_program_unit(
+                        unit_name, self._reader.filename)
```

Comment:
We should start compiling a style guide document where we agree on how we are going to handle, for instance,  continuation argument lists. Both the original (my style) and new (your style) are valid. We just need to pick one.

Reply:
I hadn't even noticed this.  I think in this case I _had_ to do it the way I have because the new longer name would pass 80 characters under the "old" style

Reply:
A discussion for another forum.